src/handlers/Priest.cpp

```diff
@@ -0,0 +1,206 @@
+
+/* $Id$
+ * EOSERV is released under the zlib license.
+ * See LICENSE.txt for more info.
+ */
+
+#include "handlers.hpp"
+
+#include "../character.hpp"
+#include "../config.hpp"
+#include "../eodata.hpp"
+#include "../map.hpp"
+#include "../npc.hpp"
+#include "../npc_data.hpp"
+#include "../packet.hpp"
+#include "../timer.hpp"
+#include "../wedding.hpp"
+#include "../world.hpp"
+
+#include "../util.hpp"
+
+namespace Handlers
+{
+
+// Accepting a marriage request
+void Priest_Accept(Character *character, PacketReader &reader)
+{
+	(void) reader;
+
+	if (!character->partner.empty())
+		return;
+
+	Wedding* wedding = character->map->wedding;
+
+	if (!wedding)
+		return;
+
+	if (wedding->Busy())
+	{
+		PacketBuilder reply(PACKET_PRIEST, PACKET_REPLY, 2);
+		reply.AddShort(PRIEST_BUSY);
+		character->Send(reply);
+	}
+	
+	Character* partner = character->world->GetCharacter(character->fiance);
+
+	if (!partner || partner->map != character->map)
+	{
+		PacketBuilder reply(PACKET_PRIEST, PACKET_REPLY, 2);
+		reply.AddShort(PRIEST_PARTNER_NOT_PRESENT);
+		character->Send(reply);
+		return;
+	}
+
+	if (character->map->wedding->RequestedWedding(partner))
+	{
+		character->map->wedding->StartWedding(partner->SourceName(), character->SourceName());
+	}
+}
+
+// Talking to a priest NPC
+void Priest_Open(Character *character, PacketReader &reader)
+{
+	short id = reader.GetShort();
+
+	if (!character->partner.empty())
+		return;
+
+	if (!character->map->wedding)
+		return;
+
+	UTIL_FOREACH(character->map->npcs, npc)
+	{
+		if (npc->index == id && npc->ENF().type == ENF::Priest)
+		{
+			if (character->map->wedding->Busy())
+			{
+				PacketBuilder reply(PACKET_PRIEST, PACKET_REPLY, 2);
+				reply.AddShort(PRIEST_BUSY);
+				character->Send(reply);
+			}
+			else
+			{
+				character->npc = npc;
+				character->npc_type = ENF::Priest;
+
+				PacketBuilder reply(PACKET_PRIEST, PACKET_OPEN, 4);
+				reply.AddInt(npc->id);
+				character->Send(reply);
+			}
+			break;
+		}
+	}
+}
+
+// Requesting marriage at a priest
+void Priest_Request(Character *character, PacketReader &reader)
+{
+	/*int session_id = */reader.GetInt();
+	reader.GetByte();
+	std::string name = reader.GetEndString();
+
+	if (character->npc_type == ENF::Priest)
+	{
+		if (character->map->wedding->Busy())
+		{
+			PacketBuilder reply(PACKET_PRIEST, PACKET_REPLY, 2);
+			reply.AddShort(PRIEST_BUSY);
+			character->Send(reply);
+		}
+
+		if (name != character->fiance || name == character->SourceName())
```

Comment:
Upon further reflection these checks should probably be case-insensitive

Reply:
I've lowercased the fiance name inputs instead, matching the actual casing of character names.
This is the approach in other places like `Account_Create`, `Guild_Player`, and `Guild_Rank`.